### ISSUE TITLE
Code: Refactor of townSpawn - townSpawn with Nation or it's allies is treated differently (Issue 250)

### ISSUE THREAD

**Issue by LlmDl** [TownyAdvanced/Towny#322]
Originally reported on Google Code with ID 286

```
New Lang:
msg_err_town_spawn_nation_forbidden: '&cTown spawn travel to other towns in your nation
is forbidden.'
msg_err_town_spawn_ally_forbidden: '&cTown spawn travel to towns allied with your nation
is forbidden.'
msg_err_public_spawn_forbidden: '&cPublic spawn travel to other towns is forbidden.'

Refactored town spawn permissions:
towny.town.spawntp.town
towny.town.spawntp.nation
towny.town.spawntp.ally
towny.town.spawntp.public

Which allows for convenience:
towny.town.spawntp.*

Basically I made a new enum TownSpawnLevel (name arguable for lameness). Check it out
for the bulk of everything.
```

Reported by `zrenfire` on 2011-09-23 04:43:40

<hr>
- _Attachment: [patch.txt](https://storage.googleapis.com/google-code-attachments/towny/issue-286/comment-0/patch.txt)_
